### ISSUE TITLE
[F-B-6] Risk Manager - Position Sizing & Risk Controls

### ISSUE THREAD

**Issue by siven76** [siven76/PA-AI#15]
## Feature Description
Implement a comprehensive risk management system with configurable per-trade risk limits, maximum simultaneous positions, portfolio-level controls, and real-time risk monitoring for safe automated trading.

## Acceptance Criteria
- [ ] Configurable per-trade risk (e.g., 1% of equity per trade)
- [ ] Maximum simultaneous positions limit
- [ ] Portfolio-level risk controls (max drawdown, exposure limits)
- [ ] Position sizing calculations based on volatility
- [ ] Stop-loss and take-profit level determination
- [ ] Real-time risk monitoring and alerts
- [ ] Emergency stop functionality
- [ ] Risk metrics calculation and reporting

## Technical Requirements
- **Framework**: Python with async support for real-time monitoring
- **Integration**: F-B-5 signal processing and F-B-7 broker bridge
- **Configuration**: YAML-based risk parameters
- **Monitoring**: Real-time portfolio tracking
- **Alerts**: Configurable risk threshold notifications
- **Persistence**: Risk metrics and trade history storage

## Risk Management Components
```python
class RiskManager:
    def __init__(self, config: RiskConfig):
        self.config = config
        self.portfolio = Portfolio()
        self.active_positions = {}
        self.risk_metrics = RiskMetrics()
        
    async def evaluate_signal(self, signal: TradingSignal) -> RiskDecision:
        """Evaluate if signal passes risk checks"""
        # 1. Portfolio-level checks
        if not self.check_portfolio_limits():
            return RiskDecision.REJECT_PORTFOLIO_LIMIT
            
        # 2. Position-level checks
        if not self.check_position_limits(signal.pair):
            return RiskDecision.REJECT_POSITION_LIMIT
            
        # 3. Calculate position size
        position_size = self.calculate_position_size(signal)
        
        # 4. Validate risk/reward ratio
        if not self.validate_risk_reward(signal, position_size):
            return RiskDecision.REJECT_RISK_REWARD
            
        return RiskDecision.APPROVE(position_size)
```

## Risk Configuration
```yaml
risk_management:
  # Per-trade risk limits
  max_risk_per_trade: 0.01  # 1% of account equity
  max_position_size: 0.05   # 5% of account equity
  
  # Portfolio limits
  max_simultaneous_positions: 5
  max_portfolio_risk: 0.10  # 10% total portfolio risk
  max_daily_loss: 0.05      # 5% daily loss limit
  max_drawdown: 0.20        # 20% maximum drawdown
  
  # Position sizing
  volatility_lookback: 20   # Days for volatility calculation
  volatility_multiplier: 2.0
  min_position_size: 0.001  # Minimum position size
  
  # Risk/Reward requirements
  min_risk_reward_ratio: 1.5
  max_stop_loss_pct: 0.02   # 2% maximum stop loss
  
  # Emergency controls
  emergency_stop_enabled: true
  max_consecutive_losses: 5
  circuit_breaker_loss: 0.10  # 10% loss triggers circuit breaker
```

## Position Sizing Algorithm
```python
def calculate_position_size(self, signal: TradingSignal) -> float:
    """Calculate optimal position size based on risk parameters"""
    
    # 1. Account-based sizing
    account_equity = self.portfolio.get_equity()
    max_risk_amount = account_equity * self.config.max_risk_per_trade
    
    # 2. Volatility-based sizing (ATR)
    atr = self.calculate_atr(signal.pair, self.config.volatility_lookback)
    volatility_adjusted_size = max_risk_amount / (atr * self.config.volatility_multiplier)
    
    # 3. Signal confidence adjustment
    confidence_multiplier = min(signal.confidence, 1.0)
    adjusted_size = volatility_adjusted_size * confidence_multiplier
    
    # 4. Apply limits
    max_position = account_equity * self.config.max_position_size
    final_size = min(adjusted_size, max_position)
    
    return max(final_size, self.config.min_position_size)
```

## Stop Loss & Take Profit Calculation
```python
def calculate_stop_levels(self, signal: TradingSignal, position_size: float) -> StopLevels:
    """Calculate stop loss and take profit levels"""
    
    entry_price = signal.current_price
    atr = self.calculate_atr(signal.pair, 14)
    
    if signal.direction == 'long':
        # Long position stops
        stop_loss = entry_price - (atr * 2.0)
        take_profit = entry_price + (atr * 3.0)  # 1.5 R:R ratio
    else:
        # Short position stops
        stop_loss = entry_price + (atr * 2.0)
        take_profit = entry_price - (atr * 3.0)
    
    # Validate stop loss doesn't exceed max percentage
    max_stop_distance = entry_price * self.config.max_stop_loss_pct
    if signal.direction == 'long':
        stop_loss = max(stop_loss, entry_price - max_stop_distance)
    else:
        stop_loss = min(stop_loss, entry_price + max_stop_distance)
    
    return StopLevels(stop_loss=stop_loss, take_profit=take_profit)
```

## Real-time Risk Monitoring
```python
class RiskMonitor:
    def __init__(self, risk_manager: RiskManager):
        self.risk_manager = risk_manager
        self.alerts = AlertManager()
        
    async def monitor_portfolio(self):
        """Continuous portfolio risk monitoring"""
        while True:
            # Calculate current risk metrics
            metrics = await self.calculate_risk_metrics()
            
            # Check for risk threshold breaches
            await self.check_risk_thresholds(metrics)
            
            # Update risk dashboard
            await self.update_risk_dashboard(metrics)
            
            await asyncio.sleep(10)  # Monitor every 10 seconds
    
    async def check_risk_thresholds(self, metrics: RiskMetrics):
        """Check if any risk thresholds are breached"""
        if metrics.current_drawdown > self.config.max_drawdown:
            await self.trigger_emergency_stop("Maximum drawdown exceeded")
            
        if metrics.daily_loss > self.config.max_daily_loss:
            await self.trigger_daily_stop("Daily loss limit exceeded")
            
        if metrics.portfolio_risk > self.config.max_portfolio_risk:
            await self.alerts.send_alert("Portfolio risk limit approached")
```

## Risk Metrics & Reporting
```python
@dataclass
class RiskMetrics:
    current_equity: float
    total_exposure: float
    portfolio_risk: float
    current_drawdown: float
    daily_pnl: float
    win_rate: float
    avg_win_loss_ratio: float
    sharpe_ratio: float
    max_consecutive_losses: int
    var_95: float  # Value at Risk (95% confidence)
    
    def to_dict(self) -> dict:
        return asdict(self)
```

## Emergency Controls
- **Circuit Breaker**: Automatic trading halt on excessive losses
- **Emergency Stop**: Manual override to close all positions
- **Position Limits**: Hard caps on position sizes and counts
- **Drawdown Protection**: Automatic risk reduction on drawdowns
- **Correlation Limits**: Prevent over-concentration in correlated pairs

## Definition of Done
- [ ] Risk management framework implementation
- [ ] Position sizing algorithms with volatility adjustment
- [ ] Stop loss and take profit calculation
- [ ] Real-time portfolio monitoring
- [ ] Risk threshold alerts and emergency stops
- [ ] Configuration management for risk parameters
- [ ] Risk metrics calculation and reporting
- [ ] Integration with signal generation and broker bridge
- [ ] Unit tests for all risk calculations
- [ ] Backtesting validation of risk controls
